velox/functions/prestosql/DateTimeImpl.h

```diff
@@ -42,13 +42,16 @@ FOLLY_ALWAYS_INLINE std::optional<Timestamp> fromUnixtime(double unixtime) {
 
   static const int64_t kMax = std::numeric_limits<int64_t>::max();
   static const int64_t kMin = std::numeric_limits<int64_t>::min();
+  static const uint64_t kSignedMax = (1ull << 63);
```

Comment:
Any particular reason not to use std::numeric_limits<uint64_t>::max();?

naming: kSignedMax is of type "unsigned int 64", e.g. the name doesn't match the value. Should it be kUnsignedMax?

Reply:
This change was a follow up from discussions here 
https://github.com/facebookincubator/velox/pull/1090#discussion_r816130583
and 
https://github.com/facebookincubator/velox/pull/1090#discussion_r816150588

The core problem is described in the first comment above.
```
If you look below at line 62, we're using floor(unixtime) to compute seconds, and in line 64 seconds gets converted to int64_t implicitly by passing it to the Timestamp constructor.

I'm trying to avoid this case where if you cast kMax to a double you actually get a number larger than kMax
https://stackoverflow.com/questions/65749055/when-a-64bit-int-is-cast-to-64bit-float-in-c-c-and-doesnt-have-an-exact-match

so if unixtime is a fractional value in between kMax and static_cast<double>(kMax) you'll miss this case
```
My understanding is that we want to represent `KSignedMax` using the unsigned type so that we can handle the cast to double correctly.